config/runtime/base/python.jinja2

```diff
@@ -111,7 +111,11 @@ def main(args):
             job.submit(results, NODE, API_CONFIG_YAML)
     except requests.exceptions.HTTPError as ex:
         print(ex, file=sys.stderr)
-        detail = ex.response.json().get('detail')
+        try:
+            detail = ex.response.json().get('detail')
+        except Exception:
+            detail = None
+            print(f"Non-JSON response, body: {ex.response}", file=sys.stderr)
```

Comment:
Sounds like the real issue here is when the detail is not JSON.  Maybe this happens when the API is returning a plain text response?

I don't think we should be handling this exception, it's just not meant to be happening and is more likely a bug in the API.  If we want a catch-all exception thing we can have it outside this whole part of the code.

Reply:
I was happening to me when one of tokens was missing, and it was really hard to troubleshoot even looking to logs. So it is definitely real issue and cover real case.

Reply:
Yes but every random failure will raise a different exception in another part of the code.  So maybe this was nice to help you debug things, I really think it should be dropped.  The real broken error handling seems to be on the API side for the reasons I've explained before.

Reply:
It might be various errors, sometimes request might not even reach python code, including 502 reponse from ingress/nginx, or something else, so we need to handle non-json response for such cases, so we dont guess what was inside the body of this reply.

Reply:
Right but that's not the same as failing to parse the JSON we're expecting from the API.  Anyway, I don't really mind...

Reply:
See #2182 (a new version of what we had in the legacy command line tools)

And yes it looks like we can't easily control the responses sent by the API when there's a problem.  We can add custom exception handlers and implement them to always return `JSONResponse` but I don't think that's really a good idea.

Maybe we could reuse the decorator in the Python job template, but not as a `kernelci.cli` decorator as that's really for the command line tools using Click and the jobs aren't in this category.  So it's probably better to just implement a similar error handling in the template for now and see if things can be consolidated later.